dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackedEntityInstanceService.java

```diff
@@ -148,9 +148,27 @@
 
         List<TrackedEntityInstance> trackedEntityInstances = trackedEntityInstanceStore.getTrackedEntityInstances( params );
 
-        for ( TrackedEntityInstance tei : trackedEntityInstances )

```

Comment:
This can be pushed down to the actual query / database, so that skip sync attributes never comes back from the API call in the first place. Will be cleaner and more performant.